File: app/src/main/java/com/quran/labs/androidquran/dao/translation/Translation.java
package com.quran.labs.androidquran.dao.translation;

public class Translation {
  public final int id;
  public final String displayName;
  public final String downloadType;
  public final String fileUrl;
  public final String saveTo;
  public final String translator;
  public final String languageCode;
  public final int minimumVersion;
  public final int currentVersion;
  public final String fileName;
  public final String translatorNameLocalized;

  public Translation(int id, int minimumVersion, int currentVersion, String displayName,
      String downloadType, String fileName, String fileUrl, String saveTo,
      String languageCode, String translator, String translatorNameLocalized) {
    this.id = id;
    this.minimumVersion = minimumVersion;
    this.currentVersion = currentVersion;
    this.displayName = displayName;
    this.downloadType = downloadType;
    this.fileName = fileName;
    this.fileUrl = fileUrl;
    this.saveTo = saveTo;
    this.languageCode = languageCode;
    this.translator = translator;
    this.translatorNameLocalized = translatorNameLocalized;
  }
}

File: app/src/main/java/com/quran/labs/androidquran/dao/translation/Translation.kt
package com.quran.labs.androidquran.dao.translation

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Translation(val id: Int,
                       val minimumVersion: Int,
                       val currentVersion: Int,
                       val displayName: String,
                       val downloadType: String,
                       val fileName: String,
                       val fileUrl: String,
                       val saveTo: String,
                       val languageCode: String,
                       val translator: String? = "",
                       @Json(name = "translatorForeign") val translatorNameLocalized: String? = "") {

  fun withSchema(schema: Int) = copy(minimumVersion = schema)
}
